python/pydantic_core/_pydantic_core.pyi

```diff
@@ -55,18 +55,43 @@ _T = TypeVar('_T', default=Any, covariant=True)
 
 @final
 class Some(Generic[_T]):
+    """
+    Similar to Rust's [`Option::Some`](https://doc.rust-lang.org/std/option/enum.Option.html) type, this
+    identifies a value as being present, and provides a way to access it.
+
+    Generally used in a union with `None` to different between "some value which could be None" and no value.
+    """
+
     __match_args__ = ('value',)
 
     @property
-    def value(self) -> _T: ...
+    def value(self) -> _T:
```

Comment:
The type annotation is `_T`, maybe we should rename to `T` for nicer docs?

![image](https://github.com/pydantic/pydantic-core/assets/1939362/bdbca953-a33d-494e-a053-7d5b62078b2c)